packages/sitecore-jss-nextjs/src/editing/editing-data-cache.ts

```diff
@@ -27,23 +27,23 @@ export class EditingDataDiskCache implements EditingDataCache {
     this.cache = new Cache('editing-data', { compression: 'gzip', location: tmpDir });
   }
 
-  set(key: string, editingData: EditingData): void {
+  async set(key: string, editingData: EditingData): Promise<void> {
     const filePath = this.cache.set(key, JSON.stringify(editingData));
     if (!filePath || filePath.length === 0) {
       throw new Error(`Editing data cache not set for key ${key} at ${this.cache.root}`);
     }
   }
 
-  get(key: string): EditingData | undefined {
+  async get(key: string): Promise<EditingData | undefined> {
     const entry = this.cache.get(key);
     if (!entry.isCached) {
       console.warn(`Editing data cache miss for key ${key} at ${this.cache.root}`);
       return undefined;
     }
-    const data = JSON.parse(entry.value);
     // Remove immediately to preserve disk-space
+    // NOTE: sync-disk-cache actually uses fs.unlink so this is async
```

Comment:
It uses `unlinkSync` which is still sync:
https://github.com/stefanpenner/sync-disk-cache/blob/master/index.js#L229
https://github.com/stefanpenner/sync-disk-cache/blob/master/index.js#L10

Reply:
🤦‍♂️Yes you're absolutely right. I was looking at [async-disk-cache](https://github.com/stefanpenner/async-disk-cache/blob/master/index.js#L12). I had at one point switched over to this, but ran into some issues (webpack warnings, plus additional dependencies) that made it less attractive. I'll revisit...

Reply:
@illiakovalenko See latest. Stuck with sync-disk-cache (and sync `EditingData[Disk]Cache`). Wrapped the cache remove in setTimeout so it's added to the macrotasks queue (non-blocking).